### PR TITLE
Add a layered extremum material property

### PR DESCRIPTION
**Pull request by GiudGiud**
Classic UO combination, but missing right now

### CODE REVIEW

File: framework/doc/content/source/userobjects/LayeredAverage.md [hunk 1]
```diff
@@ -59,7 +59,7 @@ the [!param](/UserObjects/LayeredAverage/cumulative) to `true`.
 ## Example input syntax
 
 In this example, the average of variable `u` is taken over the whole domain in direction `y` over
-two layers. The result of this averaging in stored is the variable `layered_average` using a
+two layers. The result of this averaging in stored in the variable `layered_average` using a
```
Comment (lindsayad) [inline framework/doc/content/source/userobjects/LayeredAverage.md:62]:
```suggestion
two layers. The result of this averaging is stored in the variable `layered_average` using a
```

File: framework/include/userobjects/LayeredExtremumMaterialProperty.h [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ */
+class LayeredExtremumMaterialProperty : public ElementExtremeMaterialProperty, public LayeredBase
+{
+public:
+  static InputParameters validParams();
+
+  LayeredExtremumMaterialProperty(const InputParameters & parameters);
+
+  /**
+   * Given a Point return the integral value associated with the layer that point falls in.
```
Comment (lindsayad) [inline framework/include/userobjects/LayeredExtremumMaterialProperty.h:28]:
```suggestion
   * Given a Point, return the integral value associated with the layer that point falls in.
```

File: framework/src/userobjects/LayeredExtremumMaterialProperty.C [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+void
+LayeredExtremumMaterialProperty::initialize()
+{
+  ElementExtremeMaterialProperty::initialize();
+  LayeredBase::initialize();
+
+  // Initialize layer values
+  for (auto layer : make_range(_num_layers))
```
Comment (lindsayad) [inline framework/src/userobjects/LayeredExtremumMaterialProperty.C:39]:
```suggestion
  for (const auto layer : make_range(_num_layers))
```

File: framework/src/userobjects/LayeredExtremumMaterialProperty.C [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  // Do not keep track of the layer extremum with _value
+  ElementExtremeMaterialProperty::initialize();
+  ElementExtremeMaterialProperty::execute();
+  unsigned int layer = getLayer(_current_elem->vertex_average());
+
+  setLayerValue(layer, extreme_value(getLayerValue(layer), _value));
+}
+
+Real
+LayeredExtremumMaterialProperty::extreme_value(Real a, Real b) const
```
Comment (lindsayad) [inline framework/src/userobjects/LayeredExtremumMaterialProperty.C:56]:
```suggestion
LayeredExtremumMaterialProperty::extreme_value(const Real a, const Real b) const
```

File: framework/include/userobjects/LayeredExtremumMaterialProperty.h [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+   *
+   * @param p The point to look for in the layers.
+   */
+  virtual Real spatialValue(const Point & p) const override { return integralValue(p); }
+
+  virtual const std::vector<Point> spatialPoints() const override;
+
+  virtual void initialize() override;
+  virtual void execute() override;
+  Real extreme_value(Real a, Real b) const;
```
Comment (GiudGiud) [inline framework/include/userobjects/LayeredExtremumMaterialProperty.h:38]:
```suggestion
  Real extreme_value(const Real a, const Real b) const;
```

**Review by lindsayad — APPROVED** [review]

**Comment by GiudGiud** [comment]
Test failure, kernels/ad_jacobians.testbodyforce-adfunction-jac is unrelated (incorrect jacobian by amount of numerical tolerance roughly)

**Comment by GiudGiud** [comment]
Thanks for the review!

**Pull request merged**